packages/react-magma-dom/src/components/CharacterCounter/CharacterCounter.test.js

```diff
@@ -0,0 +1,123 @@
+
+const charactersAllowed = defaultI18n.characterCounter.charactersAllowed;
+
+const characterLeft = defaultI18n.characterCounter.characterLeft;
+
+const charactersLeft = defaultI18n.characterCounter.charactersLeft;
+
+const characterOver = defaultI18n.characterCounter.characterOver;
+
+const charactersOver = defaultI18n.characterCounter.charactersOver;
+
+describe('CharacterCounter', () => {
+  it('should find element by testId', () => {
+    const { getByTestId } = render(
+      <CharacterCounter maxLength={231} testId={testId} />
+    );
+
+    expect(getByTestId(testId)).toBeInTheDocument();
+  });
+
+  it('Does not violate accessibility standards', () => {
+    const { container } = render(<CharacterCounter maxLength={22} />);
+
+    return axe(container.innerHTML).then(result => {
+      return expect(result).toHaveNoViolations();
+    });
+  });
+  describe('Titles', () => {
+    const labelText = 'Character Counter';
+
+    describe('Characters Allowed', () => {
+      it('Shows the default label of "characters allowed" if maxLength is 0', () => {
+        const { getByText } = render(
+          <I18nContext.Provider
+            value={{
+              ...defaultI18n,
+            }}
+          >
+            <CharacterCounter maxLength={0} />
+          </I18nContext.Provider>
+        );
+
+        expect(getByText('0 ' + charactersAllowed)).toBeInTheDocument();
+      });
+
+      it('Shows the default label of "character allowed" if maxLength is 1', () => {
+        const { getByText } = render(
+          <I18nContext.Provider
+            value={{
+              ...defaultI18n,
+            }}
+          >
+            <CharacterCounter maxLength={1} />
+          </I18nContext.Provider>
+        );
+
+        expect(getByText('1 ' + characterAllowed)).toBeInTheDocument();
+      });
+      it('Shows the default label of "characters allowed"', () => {
+        const { getByText } = render(
+          <I18nContext.Provider
+            value={{
+              ...defaultI18n,
+            }}
+          >
+            <CharacterCounter maxLength={2} />
+          </I18nContext.Provider>
+        );
+
+        expect(getByText('2 ' + charactersAllowed)).toBeInTheDocument();
+      });
+    });
+
+    describe('Characters Left', () => {
+      it('Shows the label "characters left" as the user types', () => {
+        const { getByText } = render(
+          <CharacterCounter inputLength={2} maxLength={4} />
+        );
+        expect(getByText('2 ' + charactersLeft)).toBeInTheDocument();
+      });
+
+      it('Shows the label "characters left" if user types to zero', () => {
+        const { getByText } = render(
+          <CharacterCounter inputLength={4} maxLength={4} />
+        );
+        expect(getByText('0 ' + charactersLeft)).toBeInTheDocument();
+      });
+
+      it('Shows the label "character left" as the user types to one remaining character', () => {
```

Comment:
Same as above, just tweak this description to refer to the component props instead of a user action